File: ios/ObjectiveC/Beacon/3-Alert/Complete/beaconAlertComplete/ViewController.m
//
//  ViewController.m
//  beaconAlertStart
//
//  Created by sarra srairi on 29/03/2016.
//  Copyright © 2016 sarra srairi. All rights reserved.
//

#import "ViewController.h"
#import "AlertViewControllerAction.h"

@interface ViewController () {
    AdtagPlaceInAppAction *currentPlaceInAppAction;
}

@end

@implementation ViewController

-(void)viewDidLoad {
    [super viewDidLoad];
 }

-(void)viewWillAppear:(BOOL)animated{
    [super viewWillAppear:animated];
    [[AdtagBeaconManager shared] registerInAppActionDelegate:self];
    [[AdtagBeaconManager shared] registerInProximityInForeground:self];
 
}
-(void)viewWillDisappear:(BOOL)animated{
    [super viewWillDisappear:animated];
    [[AdtagBeaconManager shared] unregisterInAppActionDelegate];
    [[AdtagBeaconManager shared] unregisterInProximityInForeground:self];
}

-(void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}


- (BOOL)createInAppAction:(AdtagPlaceInAppAction * _Nonnull)placeInAppAction statusManager:(id<InAppActionStatusManagerDelegate> _Nonnull)statusManager {
    if([@"popup" isEqualToString:[placeInAppAction getAction]]){
        // create a popup view
        NSLog(@"Well done! now you can create your Pop UP");
        _txtAlertMessage.text = [placeInAppAction getTitle];
        [_txtAlertMessage setNeedsDisplay];
        _buttonAlert.hidden=false;
        currentPlaceInAppAction = placeInAppAction;
        return YES ;
    }
    
    _txtAlertMessage.text = @"No Beacons ready for action";
    [_txtAlertMessage setNeedsDisplay];
    return NO ;
}

- (BOOL)removeInAppAction:(AdtagPlaceInAppAction * _Nonnull)placeInAppAction inAppActionRemoveStatus:(enum InAppActionRemoveStatus)inAppActionRemoveStatus {
    _buttonAlert.hidden= true;
    _txtAlertMessage.text = @"The In-App Action has been removed";
    [_txtAlertMessage setNeedsDisplay];
    return YES;
}

- (void) proximityContentsInForeground:(NSArray<AdtagPlaceInAppAction *> *)contents {
    for (AdtagPlaceInAppAction *place in contents) {
        NSLog(@"place: %@", place);
    }
}

- (void) prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    AlertViewControllerAction *alertViewController = (AlertViewControllerAction *) segue.destinationViewController;
    alertViewController.currentPlaceInAppAction = currentPlaceInAppAction;
}

@end

File: ios/SWIFT/Beacon/3-Alert/Complete/AlertStrategy/ViewController.swift
//
//  ViewController.swift
//  AlertStrategy
//
//  Created by sarra srairi on 11/08/2016.
//  Copyright © 2016 R&D connecthings. All rights reserved.
//

import UIKit
import ConnectPlaceActions
import AdtagAnalytics
import AdtagConnection
import AdtagLocationBeacon
import ConnectPlaceCommon

class ViewController: UIViewController, AdtagInAppActionDelegate {
    @IBOutlet weak var txtAlertMessage: UILabel!
    @IBOutlet weak var buttonInAppAction: UIButton!
    
    var currentPlaceInAppAction: PlaceInAppAction!
  
    override func viewWillAppear(_ animated: Bool) {
        super.viewDidAppear(animated);
        AdtagBeaconManager.shared.registerInAppActionDelegate(self)
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewDidAppear(animated);
        AdtagBeaconManager.shared.unregisterInAppActionDelegate()
    }
 
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    func createInAppAction(_ placeInAppAction: AdtagPlaceInAppAction, statusManager: InAppActionStatusManagerDelegate) -> Bool {
        if ("popup" == placeInAppAction.getAction()) {
            GlobalLogger.shared.debug("Well done! now you have created your alert action")
            self.txtAlertMessage.text = placeInAppAction.getTitle()
            txtAlertMessage.setNeedsDisplay()
            buttonInAppAction.isHidden = false
            currentPlaceInAppAction = placeInAppAction
            return true
        }
        self.txtAlertMessage.text = "No Beacons ready for action"
        txtAlertMessage.setNeedsDisplay()
        return false
    }
    
    func removeInAppAction(_ placeInAppAction: AdtagPlaceInAppAction, inAppActionRemoveStatus: InAppActionRemoveStatus) -> Bool {
        GlobalLogger.shared.debug("Well done! now you have removed your alert action")
        buttonInAppAction.isHidden = true
        txtAlertMessage.text = "The In-App Action has been removed"
        txtAlertMessage.setNeedsDisplay()
        return true
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if let alertViewController = segue.destination as? AlertViewControllerAction {
            if let currentAdtagPlaceInAppAction = currentPlaceInAppAction as? AdtagPlaceInAppAction {
                alertViewController.currentPlaceInAppAction = currentAdtagPlaceInAppAction
            }
        }
    }
}
